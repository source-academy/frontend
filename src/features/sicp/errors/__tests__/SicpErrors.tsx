import { mount } from 'enzyme';

import getSicpError, { SicpErrorType } from '../SicpErrors';

describe('Sicp errors:', () => {
  test('unexpected error renders correctly', () => {
    const tree = mount(getSicpError(SicpErrorType.UNEXPECTED_ERROR));
    expect(tree.debug()).toMatchSnapshot();
  });

  test('page not found error renders correctly', () => {
    const tree = mount(getSicpError(SicpErrorType.PAGE_NOT_FOUND_ERROR));
    expect(tree.debug()).toMatchSnapshot();
  });

  test('unexpected error renders correctly', () => {
    const tree = mount(getSicpError(SicpErrorType.PARSING_ERROR));
    expect(tree.debug()).toMatchSnapshot();
  });
});
