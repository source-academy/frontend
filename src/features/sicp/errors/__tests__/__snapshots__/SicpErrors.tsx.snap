// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sicp errors: page not found error renders correctly 1`] = `
"<Blueprint4.NonIdealState title=\\"Something went wrong :(\\" description={{...}} icon=\\"error\\" iconSize={48} layout=\\"vertical\\">
  <div className=\\"bp4-non-ideal-state bp4-non-ideal-state-vertical\\">
    <div className=\\"bp4-non-ideal-state-visual\\" style={{...}}>
      <Blueprint4.Icon icon=\\"error\\" size={48} aria-hidden={true} tabIndex={-1}>
        <span aria-hidden={true} icon=\\"error\\" tabIndex={-1} className=\\"bp4-icon bp4-icon-error\\" title={[undefined]}>
          <svg fill={[undefined]} data-icon=\\"error\\" width={48} height={48} viewBox=\\"0 0 20 20\\" aria-labelledby={[undefined]} role=\\"img\\">
            <path d=\\"M10 0C4.48 0 0 4.48 0 10s4.48 10 10 10 10-4.48 10-10S15.52 0 10 0zm1 16H9v-2h2v2zm0-3H9V4h2v9z\\" fillRule=\\"evenodd\\" />
          </svg>
        </span>
      </Blueprint4.Icon>
    </div>
    <div className=\\"bp4-non-ideal-state-text\\">
      <Component>
        <h4 className=\\"bp4-heading\\">
          Something went wrong :(
        </h4>
      </Component>
      <div>
        We could not find the page you were looking for. Please check the URL again. If you believe the URL is correct, kindly let us know by filing an issue at
         
        <a href=\\"https://github.com/source-academy/frontend\\">
          https://github.com/source-academy/frontend
        </a>
        .
      </div>
    </div>
  </div>
</Blueprint4.NonIdealState>"
`;

exports[`Sicp errors: unexpected error renders correctly 1`] = `
"<Blueprint4.NonIdealState title=\\"Something went wrong :(\\" description={{...}} icon=\\"error\\" iconSize={48} layout=\\"vertical\\">
  <div className=\\"bp4-non-ideal-state bp4-non-ideal-state-vertical\\">
    <div className=\\"bp4-non-ideal-state-visual\\" style={{...}}>
      <Blueprint4.Icon icon=\\"error\\" size={48} aria-hidden={true} tabIndex={-1}>
        <span aria-hidden={true} icon=\\"error\\" tabIndex={-1} className=\\"bp4-icon bp4-icon-error\\" title={[undefined]}>
          <svg fill={[undefined]} data-icon=\\"error\\" width={48} height={48} viewBox=\\"0 0 20 20\\" aria-labelledby={[undefined]} role=\\"img\\">
            <path d=\\"M10 0C4.48 0 0 4.48 0 10s4.48 10 10 10 10-4.48 10-10S15.52 0 10 0zm1 16H9v-2h2v2zm0-3H9V4h2v9z\\" fillRule=\\"evenodd\\" />
          </svg>
        </span>
      </Blueprint4.Icon>
    </div>
    <div className=\\"bp4-non-ideal-state-text\\">
      <Component>
        <h4 className=\\"bp4-heading\\">
          Something went wrong :(
        </h4>
      </Component>
      <div>
        Something unexpected went wrong trying to load this page. Please try refreshing the page. If the issue persists, kindly let us know by filing an issue at
         
        <a href=\\"https://github.com/source-academy/frontend\\">
          https://github.com/source-academy/frontend
        </a>
        .
      </div>
    </div>
  </div>
</Blueprint4.NonIdealState>"
`;

exports[`Sicp errors: unexpected error renders correctly 2`] = `
"<Blueprint4.NonIdealState title=\\"Something went wrong :(\\" description={{...}} icon=\\"error\\" iconSize={48} layout=\\"vertical\\">
  <div className=\\"bp4-non-ideal-state bp4-non-ideal-state-vertical\\">
    <div className=\\"bp4-non-ideal-state-visual\\" style={{...}}>
      <Blueprint4.Icon icon=\\"error\\" size={48} aria-hidden={true} tabIndex={-1}>
        <span aria-hidden={true} icon=\\"error\\" tabIndex={-1} className=\\"bp4-icon bp4-icon-error\\" title={[undefined]}>
          <svg fill={[undefined]} data-icon=\\"error\\" width={48} height={48} viewBox=\\"0 0 20 20\\" aria-labelledby={[undefined]} role=\\"img\\">
            <path d=\\"M10 0C4.48 0 0 4.48 0 10s4.48 10 10 10 10-4.48 10-10S15.52 0 10 0zm1 16H9v-2h2v2zm0-3H9V4h2v9z\\" fillRule=\\"evenodd\\" />
          </svg>
        </span>
      </Blueprint4.Icon>
    </div>
    <div className=\\"bp4-non-ideal-state-text\\">
      <Component>
        <h4 className=\\"bp4-heading\\">
          Something went wrong :(
        </h4>
      </Component>
      <div>
        An error occured while loading the page. Kindly let us know by filing an issue at
         
        <a href=\\"https://github.com/source-academy/frontend\\">
          https://github.com/source-academy/frontend
        </a>
         
        and we will get it fixed as soon as possible.
      </div>
    </div>
  </div>
</Blueprint4.NonIdealState>"
`;
